### ISSUE TITLE
File I/O via binding directories

### ISSUE THREAD

**Issue by lukasheinrich** [matthewfeickert/Intro-to-Docker#5]
as far as I can see, file i/o is only discussed for copying. bind mounting is also very useful

**Comment by lukasheinrich** [comment]
see: https://indico.cern.ch/event/731951/contributions/3018144/attachments/1693850/2725983/DockerAtlasSoftwareTutorial.pdf

**Comment by matthewfeickert** [comment]
Indeed. What is up now was for a 20 minute lightning (like) talk. I mentioned mounting is possible and will cover it in the next talk (which will also cause the `README` to get updated).

**Comment by dguest** [comment]
I had no idea what this is but I'm guessing it's like a symlink.

**Comment by matthewfeickert** [comment]
@dguest c.f. [slide 15](v) in @lukasheinrich's ATLAS Docker tutorial, but we'll cover [bind mounts](https://docs.docker.com/storage/bind-mounts/) and [volumes](https://docs.docker.com/storage/volumes/) in the next talk. I don't think I actually know _exactly_ what is happening in terms of the underlying file structures when a volume is mounted, but I guess that is some reading for me.

**Comment by lukasheinrich** [comment]
The filesystem that is visible in the container (starting from `/`) is the result of what's called a "union file system" which I go into a little bit more detail here:

slide 7 and 9 here. Happy to prep some more slides if that's useful

https://indico.cern.ch/event/567072/contributions/2291687/attachments/1330835/1999952/DockerASG2_Talk.pdf

**Comment by matthewfeickert** [comment]
I don't think that more slides are necessary, but if you want to take a look over the upcoming revisions that I'll make and offer your more informed feedback that would be helpful. To be clear, I think that your slides are great, but I am intentionally trying to go for a long `README` approach (or later break it up into several smaller Markdown files, each its own mini tutorial).

**Comment by lukasheinrich** [comment]
yeah I like the repo based github style -- i'd be happy to go over a new revision. lmk

**Comment by matthewfeickert** [comment]
Awesome! :D I probably won't get around to doing anything until this weekend (given other work and projects), but I'll ping you for a review once the PR is looking reasonable.
